src/routes/LostItemsContainer.js

```diff
@@ -1,23 +1,190 @@
-  }).isRequired,
-};
+  return customFilterConfig;
+}
 
-export default withStripes(LostItemsContainer);
+const filterConfig = [
+  {
+    name: ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.LOSS_TYPE],
+    cql: ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.LOSS_TYPE],
+    values: [],
+  },
+];
+
+function buildQuery(queryParams, pathComponents, resourceData, logger, props) {
+  const customFilterConfig = buildFilterConfig(queryParams.filters);
+
+  return makeQueryFunction(
+    'cql.allRecords=1',
+    SEARCH_FIELDS.map(index => `${index}="%{query.query}*"`).join(' or '),
+    {},
+    [...filterConfig, ...customFilterConfig],
+    2,
+  )(queryParams, pathComponents, resourceData, logger, props);
+}
+
+class LostItemsContainer extends React.Component {
+  static manifest = Object.freeze({
+    query: {
+      initialValue: {},
+    },
+    resultCount: {
+      initialValue: PAGE_AMOUNT,
+    },
+    resultOffset: {
+      initialValue: 0,
+    },
+    records: {
+      type: 'okapi',
+      records: 'actualCostRecords',
+      resultOffset: '%{resultOffset}',
+      perRequest: PAGE_AMOUNT,
+      path: 'actual-cost-record-storage/actual-cost-records',
+      GET: {
+        params: {
+          query: buildQuery,
+        },
+      },
+    },
+  });
+
+  static propTypes = {
+    location: PropTypes.object,
+    history: PropTypes.object,
+    mutator: PropTypes.shape({
+      records: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+      }).isRequired,
+      query: PropTypes.shape({
+        update: PropTypes.func.isRequired,
+        replace: PropTypes.func.isRequired,
+      }).isRequired,
+      resultOffset: PropTypes.shape({
+        replace: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired
+    }).isRequired,
+  }
+
+  componentDidMount() {
+    this.source = new StripesConnectedSource(this.props, this.props.stripes.logger);
+  }
+
+  componentDidUpdate() {
+    this.source.update(this.props);
+  }
+
+  onNeedMoreData = (askAmount, index) => {
+    const {
+      resultOffset,
+    } = this.props.mutator;
+
+    if (this.source) {
+      if (resultOffset && index >= 0) {
+        this.source.fetchOffset(index);
+      } else {
+        this.source.fetchMore(PAGE_AMOUNT);
+      }
+    }
+  };
+
+  querySetter = ({ nsValues, state }) => {
+    const {
+      location,
+      location: {
+        pathname,
+        search,
+      },
+      history,
+      mutator: {
+        resultOffset,
+      },
+    } = this.props;
+
+    if (nsValues.query) {
+      nsValues.query = nsValues.query.replace('*', '');
+    }
+
+    if (state.sortChanged) {
+      resultOffset.replace(0);
+    }
+
+    const url = buildUrl(location, nsValues);
+
+    if (`${pathname}${search}` !== url) {
+      history.push(url);
+    }
+  }
+
+  queryGetter = () => {
+    return get(this.props.resources, 'query', {});
```

Comment:
@Dmitriy-Litvinenko I think you can get rid of `get` here and just do: `this.props?.resources?.query ?? {}`

Reply:
Fixed